src/openrpc/media_access.json

```diff
@@ -0,0 +1,580 @@
+                    "name": "temporal-set"
+                },
+                {
+                    "name": "capabilities",
+                    "x-uses": [
+                        "xrn:firebolt:capability:media-access:volumes"
+                    ]
+                }
+            ],
+            "summary": "Access available Volumes with MediaFiles.",
+            "params": [
+                {
+                    "name": "query",
+                    "required": true,
+                    "schema": {
+                        "$ref": "#/components/schemas/Volume"
+                    }
+                }
+            ],
+            "result": {
+                "name": "volumes",
+                "schema": {
+                    "type": "array",
+                    "items": {
+                        "title": "Volume",
+                        "$ref": "#/components/schemas/Volume"
+                    }
+                }
+            },
+            "examples": [
+                {
+                    "name": "Default example #1",
+                    "params": [
+                        {
+                            "name": "query",
+                            "value": {
+                                "type": "all"
+                            }
+                        }
+                    ],
+                    "result": {
+                        "name": "volumes",
+                        "value": [
+                            {
+                                "uri": "/usb/my-usb-drive",
+                                "name": "My USB Drive",
+                                "type": "usb"
+                            },
+                            {
+                                "uri": "/local/photos",
+                                "name": "photos",
+                                "type": "local"
+                            }
+                        ]
+                    }
+                }
+            ]
+        },
+        {
+            "name": "audio",
+            "tags": [
+                {
+                    "name": "capabilities",
+                    "x-uses": [
+                        "xrn:firebolt:capability:media-access:audio"
+                    ]
+                }
+            ],
+            "summary": "Access available audio MediaFiles.",
+            "params": [
+                {
+                    "name": "volumes",
+                    "summary": "A set of queries used to match which volumes will be matched.",
+                    "required": false,
+                    "schema": {
+                        "$ref": "#/components/schemas/Volume"
+                    }
+                },
+                {
+                    "name": "files",
+                    "summary": "A set of queries used to match which files will be matched.",
+                    "required": false,
+                    "schema": {
+                        "$ref": "#/components/schemas/MediaFile"
```

Comment:
what does it mean to have both volumes and files fields passed when the Media file also gives the volume

Reply:
I actually don't understand passing MediaFile at all, should this maybe just be passing a type and not the full MediaFile object?

Reply:
it's a filter parameter. The volume parameter allows you to filter which volumes you want, using any of the properties of Volume. The MediaFile parameter does the same for files.